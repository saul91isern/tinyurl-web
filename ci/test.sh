#!/bin/sh

echo "----- Install dependencies -----"
yarn
echo "----- Linter -----"
yarn test --ci

