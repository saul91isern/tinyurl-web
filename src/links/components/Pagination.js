import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { Pagination as SemanticPagination } from "semantic-ui-react";
import { connect } from "react-redux";
import { selectPage } from "../routines";

export const Pagination = ({ totalPages, activePage, selectPage }) => {
  return (
    <SemanticPagination
      activePage={activePage}
      disabled={totalPages <= 1}
      totalPages={totalPages}
      onPageChange={(_e, { activePage }) => selectPage(activePage)}
    />
  );
};

Pagination.propTypes = {
  activePage: PropTypes.number,
  selectPage: PropTypes.func,
  totalPages: PropTypes.number,
};

const mapStateToProps = (state) => ({
  activePage: state.activePage,
  totalPages: Math.ceil(_.size(state.links || []) / state.linksPageSize),
});

const mapDispatchToProps = { selectPage };

export default connect(mapStateToProps, mapDispatchToProps)(Pagination);
