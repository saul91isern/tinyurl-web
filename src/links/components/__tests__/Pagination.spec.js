/* eslint-disable quotes */
import React from "react";
import userEvent from "@testing-library/user-event";
import { render } from "../../../test/render";
import { Pagination } from "../Pagination";

describe("<Pagination />", () => {
  it("renders Pagination", async () => {
    const totalPages = 10;
    const activePage = 1;
    const selectPage = jest.fn();
    const { queryByText, container } = render(
      <Pagination
        totalPages={totalPages}
        activePage={activePage}
        selectPage={selectPage}
      />,
      {}
    );
    expect(container).toMatchSnapshot();
    expect(queryByText(/3/)).toBeTruthy();
    userEvent.click(await queryByText(/3/));
    expect(selectPage).toHaveBeenCalledWith(3);
  });
});
