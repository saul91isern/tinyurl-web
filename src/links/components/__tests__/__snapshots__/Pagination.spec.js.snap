// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> renders Pagination 1`] = `
<div>
  <div
    aria-label="Pagination Navigation"
    class="ui pagination menu"
    role="navigation"
  >
    <a
      aria-current="false"
      aria-disabled="false"
      aria-label="First item"
      class="item"
      tabindex="0"
      type="firstItem"
      value="1"
    >
      «
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      aria-label="Previous item"
      class="item"
      tabindex="0"
      type="prevItem"
      value="1"
    >
      ⟨
    </a>
    <a
      aria-current="true"
      aria-disabled="false"
      class="active item"
      tabindex="0"
      type="pageItem"
      value="1"
    >
      1
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      class="item"
      tabindex="0"
      type="pageItem"
      value="2"
    >
      2
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      class="item"
      tabindex="0"
      type="pageItem"
      value="3"
    >
      3
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      class="item"
      tabindex="0"
      type="pageItem"
      value="4"
    >
      4
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      class="item"
      tabindex="0"
      type="pageItem"
      value="5"
    >
      5
    </a>
    <a
      aria-current="false"
      aria-disabled="true"
      class="item"
      tabindex="-1"
      type="ellipsisItem"
      value="6"
    >
      ...
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      class="item"
      tabindex="0"
      type="pageItem"
      value="10"
    >
      10
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      aria-label="Next item"
      class="item"
      tabindex="0"
      type="nextItem"
      value="2"
    >
      ⟩
    </a>
    <a
      aria-current="false"
      aria-disabled="false"
      aria-label="Last item"
      class="item"
      tabindex="0"
      type="lastItem"
      value="10"
    >
      »
    </a>
  </div>
</div>
`;
